streaming/src/test/scala/org/apache/spark/streaming/StreamingListenerSuite.scala

```diff
@@ -140,6 +141,37 @@ class StreamingListenerSuite extends TestSuiteBase with Matchers {
     }
     true
   }
+
+  test("registering listeners via spark.streaming.extraListeners") {
+    // Test for success with zero-argument constructor and sparkConf constructor
+    val conf = new SparkConf().setMaster("local").setAppName("test")
+      .set("spark.streaming.extraListeners",
+        classOf[StreamingListenerThatAcceptsSparkConf].getName + "," +
+        classOf[ReceiverInfoCollector].getName)
+    val scc = new StreamingContext(conf, Seconds(1))
+
+    scc.scheduler.listenerBus.listeners.exists {
+      _.isInstanceOf[StreamingListenerThatAcceptsSparkConf] }
+    scc.scheduler.listenerBus.listeners.exists {
+      _.isInstanceOf[ReceiverInfoCollector] }
+
+    // Test for failure with too many arguments in constructor
+    val failingConf = new SparkConf().setMaster("local").setAppName("failingTest")
+      .set("spark.streaming.extraListeners", classOf[StreamingListenerTooManyArguments].getName)
+    val thrown = intercept[SparkException] {
+      val failingScc = new StreamingContext(failingConf, Seconds(1))
+    }
+    val expectedErrorMessage =
+        "Exception when registering Streaming Listener:" +
```

Comment:
Lets not be so specific, the exact text can change. Just verify whether the message has the name of the class inside it.